dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/domain/FilePersistenceStrategyTest.kt

```diff
@@ -304,6 +306,16 @@ internal abstract class FilePersistenceStrategyTest<T : Any>(
             .isNull()
     }
 
+    @Test
+    fun `it will create and share one single writer instance`() {
+        // given
+        val persistenceStrategy = getStrategy()
+        val currentWriter = persistenceStrategy.getWriter()
+
+        // then
+        assertThat(currentWriter).isSameAs(currentWriter)
```

Comment:
Shouldn't this be:
```suggestion
        assertThat(persistenceStrategy.getWriter()).isSameAs(currentWriter)
```
?

Reply:
nope....what I want to assert is that the newly produced writer will be the same as the previous one, meaning one single instance/ strategy

Reply:
Hmm, to me, what you say is exactly what I men:
```kotlin
assertThat(persistenceStrategy.getWriter()) // newely produced writer
   .isSameAs(currentWriter) // previous one
```

Otherwise we just test if `currentWriter` object is equal to `currentWriter`, which is always `true`, no?

Reply:
oh...right...sorry for that...my bad.